ludwig/models/llm.py

```diff
@@ -66,6 +91,33 @@ def __init__(
 
         clear_data_cache()
 
+    def to_device(self, device):
+        device = torch.device(device)
+        if device == self.curr_device:
+            logger.warning(f"LLM already on device '{device}'. Skipping device placement step.")
+            return self
+        else:
+            logger.info(f"Moving LLM from '{self.curr_device}' to '{device}'.")
+
+        model_kwargs = {}
+        if device == torch.device("cuda"):
+            num_gpus = torch.cuda.device_count()
+            model_kwargs.update(
+                dict(
+                    low_cpu_mem_usage=True,
+                    torch_dtype=torch.float16,
+                    device_map="auto",
+                    max_memory={i: "13GiB" for i in range(num_gpus)},
+                )
+            )
+
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Worth adding a comment explaining why we need to do this.

Reply:
Good point. Added.